crates/sargon/src/profile/mfa/security_structures/matrices/builder/matrix_builder.rs

```diff
@@ -0,0 +1,377 @@
+#![allow(clippy::new_without_default)]
+
+use crate::prelude::*;
+
+pub type MatrixBuilderMutateResult = Result<(), MatrixBuilderValidation>;
+pub type MatrixBuilderBuildResult =
+    Result<MatrixOfFactorSourceIds, MatrixBuilderValidation>;
+
+/// Marker to distinguish between built and builder.
+#[derive(Clone, Debug, PartialEq, Eq, Hash, Serialize, Deserialize)]
+pub struct Built;
+
+pub type MatrixBuilder = AbstractMatrixBuilderOrBuilt<
+    FactorSourceID,
+    MatrixOfFactorSourceIds,
+    Built, // Marker to distinguish between built and builder.
```

Comment:
yes I saw that `**Built**` types are using `()`. But wouldn't it make more sense if `XYZBuilder` use `pub struct Builder;` while `**Built**` types use `pub struct Built;` (instead of `()`)?

Reply:
@matiasbzurovski thanks! I'm happy you stood your ground, you were right, this was bad and there was room for improvement, so you inspired me to do this:
https://github.com/radixdlt/sargon/pull/286/commits/72567130992fcbe137c1f933761c6ec51a4c84e7

I think you will like it! I managed to drop the `PhantomData` by using `const`, which are more clearly named.

Hope you like it!

> [!IMPORTANT]
> Continue your thorough reviews, always! :)